Core/Core/MobileCore.cs

```diff
@@ -274,7 +274,7 @@ public ServiceConfiguration[] GetServiceConfigurationByType(String type)
             List<ServiceConfiguration> listOfConfigs = new List<ServiceConfiguration>();
             foreach (var item in servicesConfig)
             {
-                if (item.Value.Type == type)
+                if (item.Value.Type.ToLower() == type.ToLower())
```

Comment:
I would suggest to move `type.toLower` out of the loop to avoid wasting CPU.
About item.Value.type.ToLower, I would suggest to change [here](https://github.com/aerogear/aerogear-xamarin-sdk/blob/master/Core/Core/Configuration/ServiceConfiguration.cs#L138) to save directly the type as lowercase.

If we don't want to change that, instead of lowercasing the value, use:
`
String.Equals(item.Value.Type, type, StringComparison.OrdinalIgnoreCase) 
`

Reply:
I don't think we want to save the type directly as lowercase so I'll update to use your last option, which means it will still have to be used in the loop

Reply:
@rachael-oregan : ok, let's just move the `type.toLower()` out of the loop.